tests/cli/test_cli.py

```diff
@@ -0,0 +1,505 @@
+    return CliRunner()
+
+
+def create_example_project(temp_dir) -> None:
+    """
+    Sets up CLI tests requiring a real SQLMesh project by:
+        - Creating the SQLMesh example project in the temp_dir directory
+        - Overwriting the config.yaml file so the duckdb database file will be created in the temp_dir directory
+    """
+    init_example_project(temp_dir, "duckdb")
+    with open(temp_dir / "config.yaml", "w") as f:
+        f.write(
+            f"""gateways:
+    local:
+        connection:
+            type: duckdb
+            database: {temp_dir}/db.db
+
+default_gateway: local
+
+model_defaults:
+    dialect: duckdb
+"""
+        )
+
+
+def update_incremental_model(temp_dir) -> None:
+    with open(temp_dir / "models" / "incremental_model.sql", "w") as f:
+        f.write(
+            """
+MODEL (
+    name sqlmesh_example.incremental_model,
+    kind INCREMENTAL_BY_TIME_RANGE (
+        time_column ds
+    ),
+    start '2020-01-01',
+    cron '@daily',
+    grain (id, ds)
+);
+
+SELECT
+    id,
+    item_id,
+    'a' as new_col,
+    ds,
+FROM
+    sqlmesh_example.seed_model
+WHERE
+    ds between @start_ds and @end_ds
+"""
+        )
+
+
+def update_full_model(temp_dir) -> None:
+    with open(temp_dir / "models" / "full_model.sql", "w") as f:
+        f.write(
+            """
+MODEL (
+  name sqlmesh_example.full_model,
+  kind FULL,
+  cron '@daily',
+  grain item_id,
+  audits (assert_positive_order_ids),
+);
+
+SELECT
+  item_id,
+  count(distinct id) AS num_orders,
+FROM
+    sqlmesh_example.incremental_model
+GROUP BY item_id
+"""
+        )
+
+
+def assert_duckdb_test(result) -> None:
+    assert "Successfully Ran 1 tests against duckdb" in result.output
+
+
+def assert_new_env(result, new_env="prod", from_env="prod") -> None:
+    assert f"New environment `{new_env}` will be created from `{from_env}`" in result.output
+
+
+def assert_model_versions_created(result) -> None:
+    assert "All model versions have been created successfully" in result.output
+
+
+def assert_model_batches_executed(result) -> None:
+    assert "All model batches have been executed successfully" in result.output
+
+
+def assert_target_env_updated(result) -> None:
+    assert "The target environment has been updated successfully" in result.output
+
+
+def assert_backfill_success(result) -> None:
+    assert_model_versions_created(result)
+    assert_model_batches_executed(result)
+    assert_target_env_updated(result)
+
+
+def assert_plan_success(result, new_env="prod", from_env="prod") -> None:
+    assert result.exit_code == 0
+    assert_duckdb_test(result)
+    assert_new_env(result, new_env, from_env)
+    assert_backfill_success(result)
+
+
+def assert_virtual_update(result) -> None:
+    assert "Virtual Update executed successfully" in result.output
+
+
+def test_version(runner):
+    result = runner.invoke(cli, ["--version"])
+    assert result.exit_code == 0
+    assert re.fullmatch("\d+\.\d+\.\d+(\..*)?\n?", result.output)
```

Comment:
Don't we know what version should be reported based on the value in the version file? I'm wondering if we should assert we get that as a response instead of the softer assert that we get something that matches a regex pattern.